src/pages/index.js

```diff
@@ -5,28 +5,78 @@ import Section from "../components/Section.js";
 import {
   initialCards,
   selectors,
-  addCardImgUrlInput,
-  addCardTitleInput,
+  imageProfileAvatar,
   nameInput,
   jobInput,
   ImagePopupSelector,
   addCardButton,
   config,
   addCardFormElement,
+  editAvatarFormElement,
   editProfileFormElement,
   editProfileButton,
 } from "../utils/Constants.js";
 import Card from "../components/Card.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
+import { Api } from "../components/Api.js";
+import DeleteConfirmationPopup from "../components/DeleteConfirmationPopup.js";
 
 const handleImageClick = (data) => {
   imagePopup.open(data);
 };
 
 const createCard = (cardData) => {
-  const card = new Card(cardData, selectors.Cardtemplate, handleImageClick);
+  const card = new Card(
+    cardData,
+    selectors.Cardtemplate,
+    handleImageClick,
+    function handleDeleteModal() {
+      deleteCardForm.open();
+
+      deleteCardForm.setSubmitAction(() => {
+        api
+          .deleteCard(card.id)
+          .then((res) => {
+            console.log("The delete card function is being called");
+            // delete the card element
+            card.deleteCard();
+            deleteCardForm.renderLoading(true, "Deleting...");
+            deleteCardForm.close();
+          })
+          .finally(() => {
+            deleteCardForm.renderLoading(true, "Yes");
+          })
+          .catch((res) => {
+            console.log(`There is an error in the program: ${res}`);
+          });
+      });
+    },
+    function addLikeButton() {
+      api
+        .addLike(card.id)
+        .then((res) => {
+          card.isLiked = res.isLiked;
+          card.addLikeButtonElement();
+        })
+        .catch((res) => {
+          console.log(`There is an error in the program: ${res}`);
+        });
+    },
+    function removeLikeButton() {
+      api
+        .removeLike(card.id)
+        .then((res) => {
+          card.isLiked = res.isLiked;
+          card._removeLikeButtonElement();
```

Comment:
Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.